src/libs/structs/state.rs

```diff
@@ -72,6 +74,31 @@ pub struct StrataState<BackendData: Backend + 'static> {
 	pub pointer_location: Point<f64, Logical>,
 }
 
+pub struct GlobalState<BackendData: Backend + 'static> {
+	inner: OnceCell<Mutex<StrataState<BackendData>>>,
+}
+
+impl<BackendData: Backend + 'static> GlobalState<BackendData> {
+	pub fn new() -> Self {
+		Self { inner: OnceCell::new() }
+	}
+
+	pub fn set(&self, state: StrataState<BackendData>) -> Result<(), String> {
+		match self.inner.set(Mutex::new(state)) {
+			Ok(_) => Ok(()),
+			Err(_) => Err("Failed to set StrataState in GlobalState".to_string()),
+		}
+	}
+
+	pub fn get(&self) -> std::sync::MutexGuard<'_, StrataState<BackendData>> {
+		self.inner.get().expect("State not initialized").lock().expect("Failed to lock state")
+	}
+
+	pub fn get_mut(&self) -> std::sync::MutexGuard<'_, StrataState<BackendData>> {
+		self.inner.get().expect("State not initialized").lock().expect("Failed to lock state")
+	}
```

Comment:
```suggestion
	pub fn get(&self) -> Mutex<StrataState<BackendData>> {
		self.inner.get()
	}
```

No need to panic here, maybe we'll have some use for the results when we call these explicitly. And also probably no need for a mut version in the end.